### ISSUE TITLE
Minetest Game fails to generate world using singlenode

### ISSUE THREAD

**Issue by benjaminpants** [minetest/minetest#11544]
##### Minetest version
```
Minetest 5.4.1 (Windows)
```

##### OS / Hardware

Operating system: Windows 10 (Version 2004)
CPU: AMD Ryzen 5 3600 6-Core Processor 3.59 GHz

##### Summary

Whenever creating a singlenode world in Minetest game, the butterflies mod errors out with the following message:

![image](https://user-images.githubusercontent.com/27522994/129622766-32d90f8f-abab-462a-af3f-5e4ed42491ee.png)

It's to note that this only happens once upon a world's initial creation, and sometimes doesn't happen at all.

##### Steps to reproduce

1. Download a fresh copy of Minetest
2. Create a new singlenode world
3. Attempt to play the world, the butterflies mod will give you the above error.
(This bug hasn't currently been replicated yet however, even with the steps above)

**Comment by wsor4035** [comment]
can reliably create this error with https://gist.github.com/wsor4035/8f749c977db5d7853cc945139d787fcb on windows, with backend set to dummy, error is random on sqlite backend. platform is windows

**Comment by appgurueu** [comment]
Does it have any effect if you change the number of mapgen threads?

**Comment by appgurueu** [comment]
If you get this error, `EmergeManager::getCurrentMapgen` must return `null`. This happens either if (1) no threads are active, which is impossible unless you were calling `get_mapgen_object()` outside of `register_on_generated` (which is not the case) or (2) as a fallthrough if the search condition of `t->isRunning() && t->isCurrentThread()` is not met. I'm pretty sure that `t` must be running if it is the current thread; therefore, I assume `t->isCurrentThread()` evaluates to `false` for all threads (perhaps because the callback is run in another thread?). The used thread library contains a couple quirks regarding Windows, so that looks like a start to me.

**Comment by wsor4035** [comment]
> Does it have any effect if you change the number of mapgen threads?

nope. (tested using my minekart game with backend set to dummy as the gist is just code derived from it)

**Comment by LoneWolfHT** [comment]
I've had this issue with two game jam games so far

* https://content.minetest.net/packages/Droog71/moontest/ (half-fixed after I reset `num_emerge_threads`?)
* https://content.minetest.net/packages/AFCM/subway_miner/ (always crashes, no matter what `num_emerge_threads` is set to)

**Comment by sfan5** [comment]
Tried the latter one and couldn't reproduce it (again) so I'll mark this as Windows-only bug.

**Comment by LoneWolfHT** [comment]
This is caused when Minetest triggers on_generated right as the world loads in, I can fix it by waiting ~1 second after mods are loaded before I allow get_mapgen_object() to be called
